lib/middleware/sidekiq/only_run_while_up.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Middleware
+  module Sidekiq
+    require 'down_time_checker'
+    class OnlyRunWhileUp
+      def call(worker, job, _queue)
+        if worker.respond_to?(:downtime_checks)
+          array = worker.downtime_checks.map { |service| DownTimeChecker.new(service).down? }
+          if array.all? { |element| element == false }
+            yield
+          else
+            worker.class.perform_in(array.select(&:itself).max, job['args'])
```

Comment:
I'm not sure I understand why `select(&:itself)` is necessary here. Could you explain please?

Reply:
I sorta began to build it for future use to expand to more than just checking one service.

in that scenario, you'd possibly have `[false, 600, 1800]`. This selects all the truthy's.